### DISCUSSION TITLE
[Feature] Rethink server setup

### DISCUSSION THREAD

**Discussion by ann0see** [jamulussoftware/jamulus#1137] [Documentation]
This should be worked on AFTER we merged #101 

**Is your feature request related to a problem? Please describe.**
Currently the server setup is too wordy and (at least in my opinion) not easy to follow. We should rethink it in a "do this and that" style, especially the private server setup is a bit confusing since it doesn't really explain what to do. Setting up a server isn't as difficult as depicted there.

**Describe the solution you'd like**
Rethink the server setup; focus more on setting up a private server.

1. Remove the servertypes page and integrate it in the setting up a server page
2. Add information that you can either use the server icon on macOS/Windows or use the commandline (especially on Linux).

#45 #50

**Comment by gilgongo** [comment]
Regarding 2: mentioning the command line in the presence of Windows or Mac users is the kiss of death. Do not do it :-)

BTW One thing I think we need to examine is a way of helping people decide whether they should set up a public server. I think there may be a vicious cycle going on brought about by having too many empty servers on line, leading to less use of Jamulus than there could be because nobody wants to join an empty server. So maybe if we encourage people to look at explorer.jamulus.io to find out whether there are any servers near them and if there are, think twice about setting up another one? I've taken down my own London-based server for that reason. 

Also, the same goes for private servers - there is a semi-conscious reason why the server docs are a bit long and complicated, and that's because you don't need to set one up. So it's sort of trying to put you off. Most people really should be testing things out on public servers, but there seems to be a mistaken feeling that setting up a server will make for lower latency sessions. That's just not true for 90% people who would use Jamulus with a few friends, and there are some non-obvious reasons why going for a private server may in fact mean you get worse performance - for example if you make band members play on it when they could find a public server that's got better latency. 

It's a tricky area...

**Comment by gilgongo** [comment]
> This should be worked on AFTER we merged #101

Why do you say that?

**Comment by ann0see** [comment]
> Why do you say that?

Since I think we should focus on doing one by one (and this is already almost finished)

**Comment by ann0see** [comment]
Ok. Have some thoughts on this:

We might put the focus more on setting up private/semi private servers (unregister after you've registered your server). Linux headless script could for example not default for a public server.

**Comment by gilgongo** [comment]
Possibly. But it's hard to untangle the issues with all this. And it's just a hunch that the reason for so many empty servers is down to the vicious cycle I mentioned. Perhaps there's another reason for that? And whether it damages the perception of Jamulus is hard to say too.

I guess we need to try to pin down what we want to achieve in re-writing the doc (other than just make it read nicer).

**Comment by ann0see** [comment]
> I guess we need to try to pin down what we want to achieve in re-writing the doc (other than just make it read nicer).

Yes. 

What I think (at least that's my pov):

Client:
1. Jamulus is not too difficult to use once it's setup
2. The documentation should not evoke the impression that Jamulus is difficult to setup
3. It should tell the user that he can start right away if he doesn't have an audio interface/LAN but he will suffer from audio issues (you could be up and running with your old laptop with wifi and ASIO4ALL)
3. The documentation should also help a user who doesn't really know how to install software to setup Jamulus

Server:

We have two types of people here:

1. The ones who know how to use PCs, Servers,... (they will probably use headless servers etc.) --> they should be able to get all the information they need as fast as possible
2. The ones who don't know much about PCs (they will probably also not know about port forwarding,...) --> they need some more help (and will probably not see the Google Survey since they're inexperienced)

Concerning the empty servers: Maybe they are just "copy and paste" results on Linux?

**Comment by gilgongo** [comment]
OK - I'll try a re-draft along those lines and see what it feels like.

> Maybe they are just "copy and paste" results on Linux?

It's true that about 70% of servers are on Linux, of which I would expect only a very small number (if any) are running the desktop GUI. But if they are "copy/paste" then what are the owners intending? "I'll just read this long document, set up this systemd unit file and leave a Jamulus server running because..."? I think it's more likely they want to "contribute" a public server but don't realise that if that means there are (for example) 6 servers in San Francisco, then they're more likely to all be empty than if there's only 2. So I think the server docs should mention this. Perhaps in the form of a sort of "decision tree". 

But I'll have a think,

**Comment by ann0see** [comment]
Actually during my research on the internet I found many pages linking to the server setup page. At least my feeling was that many in fact would like to have private server since they use Jamulus for rehearsals in their groups (often choirs)?

**Comment by gene96817** [comment]
I would like to tie this back to a comment made (somewhere else, I don't remember where) that there are many empty servers or seldom used servers running.

1-  Running a private server, especially on a cloud service, is to minimize cost to sessions for the owner,.
2- A private server is one method to avoid interruption of a rehearsal
3- Authentication is one way to solve
        -- privacy of a session
        -- making the server private to a closed group

I would like to see a thoughtful way to use authentication to enable as many servers as possible to be open to the community when their owners are not using the servers. It would be really great to keep all the idle servers time available to the greater community.
